### PR TITLE
Code Quality: Refactored ToolbarViewModel

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

Refactored ToolbarViewModel and separated it into ToolbarViewModel and PathBreadcrumbViewModel.

## Task Checklist

- [x] Refactored ToolbarViewModel
- [x] Add PathBreadcrumbViewModel
- [ ] Move code from ToolbarViewModel
- [ ] Clean up
- [ ] Apply requested changes

## Known Issues

_N/A_

## PR Checklist

- [x] **Closes**: _N/A_
- [ ] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Steps To Validate Changes

1. Made window small and confirmed the search box visual states are working correctly, clicking the small search button changes the path bar into a search box.
2. Confirmed the command palette can be toggled by typing `>` and using the keyboard shortcut.
3. Confirmed the styling hasn't changed
4. Confirmed the breadcrumb drop downs work as expected

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

File: src/Files.App/Data/Items/PathBreadcrumbItem.cs [hunk 1]
```diff
@@ -3,9 +3,9 @@
 
 namespace Files.App.Data.Items
 {
-	public class PathBoxItem
+	public class PathBreadcrumbItem
 	{
-		public string? Title { get; set; }
+		public string? Name { get; set; }
```
Comment (yaira2) [inline src/Files.App/Data/Items/PathBreadcrumbItem.cs:8]: How about "FolderName"?

Comment (0x5bfa) [reply src/Files.App/Data/Items/PathBreadcrumbItem.cs:8]: We may use somewhere else (eg. nested settings page) and who knows it, so I’d like to keep this general. But I’m fine if you are ok with FolderName.

Comment (0x5bfa) [reply src/Files.App/Data/Items/PathBreadcrumbItem.cs:8]: ItemName

Comment (yaira2) [reply src/Files.App/Data/Items/PathBreadcrumbItem.cs:8]: I'm not sure if we'll use this anywhere else, but we can keep it generic just in case.

**Comment by yaira2** [comment]
Code wise this looks good, but we'll need to do a lot of testing.

File: src/Files.App/Views/MainPage.xaml.cs [hunk 2]
```diff
@@ -112,15 +100,6 @@ private async Task AppRunningAsAdminPromptAsync()
 				UserSettingsService.ApplicationSettingsService.ShowRunningAsAdminPrompt = false;
 		}
 
-		// WINUI3
-		private ContentDialog SetContentDialogRoot(ContentDialog contentDialog)
```
Comment (yaira2) [inline src/Files.App/Views/MainPage.xaml.cs:116]: This is required to prevent a crash

Comment (0x5bfa) [reply src/Files.App/Views/MainPage.xaml.cs:116]: I can build on my end and tested all stuff, and CI even fails to build.

Comment (yaira2) [reply src/Files.App/Views/MainPage.xaml.cs:116]: The CI wouldn't catch this as `PromptForReviewAsync` is only called in certain cases.

Comment (0x5bfa) [reply src/Files.App/Views/MainPage.xaml.cs:116]: But this is used from nowhere.

Comment (yaira2) [reply src/Files.App/Views/MainPage.xaml.cs:116]: Should be fine in this case 👍

**Comment by 0x5bfa** [comment]
Can you merge?

**Comment by yaira2** [comment]
Is this ready for review? You still have it marked as a draft.

**Comment by 0x5bfa** [comment]
Yes

File: src/Files.App/UserControls/AddressToolbar.xaml.cs [hunk 3]
```diff
@@ -93,43 +113,61 @@ private void VisiblePath_KeyDown(object _, KeyRoutedEventArgs e)
...
 		private void OngoingTasksActions_ProgressBannerPosted(object? _, StatusCenterItem e)
 		{
-			if (OngoingTasksViewModel is not null)
-				OngoingTasksViewModel.NewItemAdded -= OngoingTasksActions_ProgressBannerPosted;
+			if (StatusCenterViewModel is not null)
+				StatusCenterViewModel.NewItemAdded -= OngoingTasksActions_ProgressBannerPosted;
 
-			// Displays a teaching tip the first time a banner is posted
-			if (userSettingsService.AppSettingsService.ShowStatusCenterTeachingTip)
+			// Displays a teaching tip at the first time a banner is posted
```
Comment (yaira2) [inline src/Files.App/UserControls/AddressToolbar.xaml.cs:166]:
```suggestion
			// Displays a teaching tip the first time a banner is posted
```

File: src/Files.App/UserControls/PathBreadcrumb.xaml [hunk 4]
```diff
@@ -44,35 +44,36 @@
 
 					<!--  Header  -->
 					<TextBlock
+						Margin="0,0,4,0"
 						FontSize="12"
-						FontWeight="Medium"
-						Text="{x:Bind Title, Mode=OneWay}" />
+						Style="{StaticResource BodyStrongTextBlockStyle}"
```
Comment (yaira2) [inline src/Files.App/UserControls/PathBreadcrumb.xaml:49]: Is the style the same as before?

File: src/Files.App/UserControls/PathBreadcrumb.xaml [hunk 5]
```diff
@@ -44,35 +44,36 @@
...
+					DragOver="PathBreadcrumbItem_DragOver"
+					Drop="PathBreadcrumbItem_Drop"
 					Orientation="Horizontal"
 					Spacing="4"
 					Tag="{x:Bind Path, Mode=OneWay}">
 
 					<!--  Header  -->
 					<TextBlock
-						Padding="0,4"
+						Margin="0,4"
```
Comment (yaira2) [inline src/Files.App/UserControls/PathBreadcrumb.xaml:72]: This change will require some testing

**Review by yaira2 — APPROVED** [review]

**Pull request merged**